### ISSUE TITLE
Reverse text doesn't work properly on Unicode texts

### ISSUE THREAD

**Issue by whysodrooled** [gemidyne/microtf2#199]
Hello, I've just checked out that there was a bug that Unicode characters wasn't properly reversed.
It shows the ? characters and some garbled texts.

I got the temporary workaround for this one just in case:
```sourcemod
// unicode support
if(c >= 1 && buffer[c-1] & 0xC0 == 0xC0) // 2 bytes
{
    for(int j = 0; j < 2; j++)
    {
        rewritten[rc] = buffer[c-1+j];
        rc++;
    }
    c--;
}
else if(c >= 2 && buffer[c-2] & 0xE0 == 0xE0) // 3 bytes
{
    for(int j = 0; j < 3; j++)
    {
        rewritten[rc] = buffer[c-2+j];
        rc++;
    }
    c -= 2;
}
else
{
    rewritten[rc] = buffer[c];
    rc++;
}
```

Hoping the bug is fixed on this one since the game uses translations. Thanks!

EDIT: Just added to check if the array number is valid. No wonder why server lagged hell when I was testing. Sorry.

**Comment by stevu236** [comment]
Thanks - I'll get this added in and tested. I might back port this to the stable version too since the v5 will take a while longer before release

**Comment by stevu236** [comment]
Hi,

We have tested & released the fix in [v4.2.0](https://github.com/gemidyne/microtf2/releases/tag/v4.2.0) - it has been merged up into the v5 dev branch too.
 
Thanks for your contribution to the gamemode & my apologies for the fix not making it out sooner.
